src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -110,13 +113,26 @@ public ReadsSparkSource(JavaSparkContext ctx) {
      * @return the header for the bam.
      */
     public static SAMFileHeader getHeader(final JavaSparkContext ctx, final String filePath) {
-        final SAMFileHeader samFileHeader;
         try {
-             samFileHeader = SAMHeaderReader.readSAMHeaderFrom(new Path(filePath), ctx.hadoopConfiguration());
+            FileSystem fs = FileSystem.get(ctx.hadoopConfiguration());
+            Path path = fs.makeQualified(new Path(filePath));
+            if (fs.isDirectory(path)) {
+                FileStatus[] bamFiles = fs.listStatus(path, new PathFilter() {
+                    private static final long serialVersionUID = 1L;
+                    @Override
+                    public boolean accept(Path path) {
+                        return path.getName().startsWith("part-");
+                    }
+                });
+                if (bamFiles.length == 0) {
+                    throw new GATKException("No BAM files to load header from in: " + path);
```

Comment:
This should probably be a UserException rather than a GATKException since it's almost certainly from someone specifying a bad path.